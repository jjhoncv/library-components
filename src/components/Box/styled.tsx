import styled from 'styled-components';

export const Box = styled.div`
  display: flex;
  flex-wrap: wrap;
  padding: 34px 34px 0;
  width: 100%;
  position: relative;
  z-index: 1;
`;
