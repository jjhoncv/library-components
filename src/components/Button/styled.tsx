import styled, { keyframes } from 'styled-components';
import { COLORS, FONTS } from './../../ui'

interface Props {
  fullWidth?: boolean;
  disabled?: boolean;
  loading?: boolean;
  primary?: boolean;
}

const animationLoading = keyframes`
  to {transform: rotate(360deg);}
`;

export const ButtonLoading = styled.div`
  display: inline-block;
  box-sizing: border-box;
  width: 20px;
  height: 20px;
  border-radius: 50%;
  border: 2px solid ${COLORS.BLACK};
  border-top: 2px solid ${COLORS.LIGHT_GRAY};
  animation: ${animationLoading} .6s linear infinite;

  ${(props: Props) => props.primary && `
    border: 2px solid ${COLORS.WHITE};
    border-top: 2px solid ${COLORS.BLACK};
  `};
`;

export const ButtonStyled = styled.button`
  display: flex;
  justify-content: center;
  align-items: center;
  width: ${(props: Props) => props.fullWidth ? '100%' : 'auto'};
  background-color: ${COLORS.LIGHT_GRAY};
  height: 44px;
  padding: 12px 25px;
  border: none;
  border: 1px solid ${COLORS.OPAQUE_GRAY};
  border-radius: 4px;
  color: ${COLORS.BLACK};
  font-size: 14px;
  font-family: ${FONTS.ROBOTO_REGULAR};
  cursor: pointer;

  &:hover {
    background-color: ${COLORS.HOVER_GRAY};
  }

  ${(props: Props) => props.disabled && `
    pointer-events: none;
    user-select: none;
    opacity: 0.3;
  `};

  ${(props: Props) => props.loading && `
    pointer-events: none;
    user-select: none;
  `};

  ${(props: Props) => props.primary && `
    background:${COLORS.BLACK};
    color: ${COLORS.WHITE};

    &:hover {
      background-color: ${COLORS.HOVER_BLACK};
    }
  `};
`;
