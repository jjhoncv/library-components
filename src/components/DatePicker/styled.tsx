import styled from 'styled-components';
import { COLORS, FONTS } from './../../ui';

interface Props {
  width?: string;
  fullWidth?: boolean;
  error?: boolean;
  disabled?: boolean;
}

interface WeekDateProps {
  disabled?: boolean;
  selected?: boolean;
}

export const DatePickerWrapper = styled.div`
  display: inline-flex;
  flex-wrap: wrap;
  width: ${(props: Props) => props.fullWidth ? '100%' : props.width};
  font-family: ${FONTS.ROBOTO_REGULAR};

  ${(props: Props) => props.disabled && `
    pointer-events: none;

    ${DatePickerBox} {
      opacity: 0.3;
    }
  `}
`;

export const DatePickerIcon = styled.div`
  display: flex;
  flex-wrap: wrap;
  position: absolute;
  right: 8px;
  top: 8px;
  cursor: pointer;
  z-index: 1;

  ${(props: Props) => props.error && `
    svg {
      color: black;
    }
  `}
`;

export const DatePickerBox = styled.div`
  display: inline-flex;
  flex-wrap: wrap;
  align-items: center;
  width: 100%;
  position: relative;
  z-index: 0;
  color: ${COLORS.BLACK};
  font-size: 14px;
`;

export const DatePickerInput = styled.input`
  background-color: transparent;
  width: 100%;
  height: 40px;
  padding: 12px 30px 12px 12px;
  border: 2px solid ${COLORS.BORDER_GRAY};
  box-sizing: border-box;
  border-radius: 4px;
  cursor: pointer;
  z-index: 2;
  &::placeholder {
    font-family: ${FONTS.ROBOTO_REGULAR};
  }
  &:focus {
    outline: none;
    border: 2px solid ${COLORS.AQUA_BLUE};
  }

  ${(props: Props) => props.error && `
    border: 2px solid ${COLORS.ERROR_RED};
  `}

  &:focus ~ ${DatePickerIcon} {
    svg {
      color: black;
    }
  }
`;

export const DatePickerCalendarWrapper = styled.div`
  position: absolute;
  margin-top: 40px;
  z-index: 9;
  width: 240px;
  max-height: 300px;
  background-color: ${COLORS.WHITE};
  color: ${COLORS.BLACK};
  border: 1px solid ${COLORS.BORDER_GRAY};
  box-shadow: 0px 0px 8px ${COLORS.HOVER_GRAY};
  border-radius: 4px;
  padding: 5px;
`;

export const DatePickerMonth = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  width: 100%;
  font-weight: bold;
`;

export const DatePickerPrev = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex: 1 1 0;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  transition: all 0.3s linear;

  &:hover {
    background: ${COLORS.AQUA_BLUE};
    color: ${COLORS.WHITE};
  }
`;

export const DatePickerNext = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex: 1 1 0;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  transition: all 0.3s linear;

  &:hover {
    background: ${COLORS.AQUA_BLUE};
    color: ${COLORS.WHITE};
  }
`;

export const DatePickerTitle = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex: 5 1 0;
  justify-content: center;
`;

export const DatePickerWeekDays = styled.div`
  display: flex;
  flex-wrap: wrap;
  width: 100%;
`;

export const DatePickerWeekDay = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex: 1 1 0;
  justify-content: center;
  align-items: center;
  font-weight: bold;
`;

export const DatePickerDates = styled.div``;

export const DatePickerWeek = styled.div`
  display: flex;
  flex-wrap: wrap;
`;

export const DatePickerWeekDate = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex: 1 1 0;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  transition: all 0.3s linear;

  ${(props: WeekDateProps) => props.disabled && `
    color: ${COLORS.TEXT_GRAY} !important;
    cursor: not-allowed;

    &: hover {
      background: ${COLORS.WHITE} !important;
      color: ${COLORS.TEXT_GRAY} !important;
    }
  `}

  ${(props: WeekDateProps) => props.selected && `
    background-color: ${COLORS.AQUA_BLUE};
    color: ${COLORS.WHITE};
  `}

  &:hover {
    background: ${COLORS.AQUA_BLUE};
    color: ${COLORS.WHITE};
  }
`;


