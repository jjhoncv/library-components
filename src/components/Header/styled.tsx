import styled from 'styled-components';
import { COLORS, FONTS } from './../../ui';

export const HeaderWrapper = styled.div`
  width: 100%;
`;
export const HeaderTitle = styled.span`
  display: block;
  font-family: ${FONTS.MULI_BOLD};
  font-size: 24px;
  line-height: 30px;
  padding-bottom: 7px;
  border-bottom: 1px black dashed;
  color: ${COLORS.TEXT_GRAY};
`;
export const HeaderSubTitle = styled.span`
  display: block;
  font-family: ${FONTS.MULI_REGULAR};
  font-size: 18px;
  line-height: 23px;
  color: ${COLORS.BLACK};
  padding-top: 9px;
`;

export const HeaderTitleImage = styled(HeaderTitle)`
  display: flex;
  ${({ pos }) => (pos === "start") && `
    justify-content: flex-end;
    flex-direction: row-reverse;
    align-items: flex-end;
  `}

  ${({ pos }) => (pos === "end") && `
    align-items: flex-end;
    justify-content: space-between;
  `}
`;
