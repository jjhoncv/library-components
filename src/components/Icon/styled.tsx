import styled from 'styled-components';
import { COLORS } from './../../ui';

interface Props {
  color?: string,
  size?: string,
  colorHover?: string
}

export const IconStyled = styled.div`
  display: inline-flex;
  flex-wrap: wrap;

  svg {
    color: ${(props: Props) => props.color || `${COLORS.TEXT_GRAY}`}
    ${(props: Props) => `
      width: ${props.size || 'auto'};
      height: ${props.size || 'auto'};
    `}
  }

  &:hover {
    svg {
      color: ${(props: Props) => props.colorHover && `${props.colorHover}`};
    }
  }
`;
