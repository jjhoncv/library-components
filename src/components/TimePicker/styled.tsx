import styled from 'styled-components';
import { COLORS, FONTS } from './../../ui';

interface Props {
  width?: string;
  fullWidth?: boolean;
  error?: boolean;
  disabled?: boolean;
}

export const TimePickerStyled = styled.div`
  display: inline-flex;
  flex-wrap: wrap;
  width: 100%;
  position: relative;
`;

export const TimePickerInput = styled.input`
  width: 100%;
  background-color: ${COLORS.WHITE};
  border: none;
  border: 2px solid ${COLORS.BORDER_GRAY};
  border-radius: 4px;
  height: 40px;
  padding: 12px;
  cursor: pointer;
  outline: 0;
  padding: 12px 28px 12px 12px;

  ${(props: Props) => props.error && `
    border: 2px solid ${COLORS.ERROR_RED};
  `}

  &::placeholder{
    color: ${COLORS.TEXT_GRAY};
    font-family: ${FONTS.ROBOTO_REGULAR}
  }

  &:focus {
    border: 2px solid ${COLORS.AQUA_BLUE};
  }
`;

export const TimePickerIcon = styled.div`
  display: flex;
  flex-wrap: wrap;
  position: absolute;
  right: 8px;
  top: 8px;
  cursor: pointer;
  z-index: 1;
`;

export const TimePickerList = styled.ul`
  position: absolute;
  background-color: ${COLORS.WHITE};
  border: 1px solid black;
  width: 100%;
  height: 100px;
  border-radius: 4px;
  border: 1px solid ${COLORS.BORDER_GRAY};
  box-shadow: 0px 0px 8px ${COLORS.HOVER_GRAY};
  overflow-y: auto;
  z-index: 9;
`;
export const TimePickerItem = styled.li`
  padding: 0 12px;
  cursor: pointer;

  &:hover {
    background: ${COLORS.AQUA_BLUE};
    color: ${COLORS.WHITE};
  }
`;

export const TimePickerWrapper = styled.div`
  display: inline-flex;
  flex-wrap: wrap;
  width: ${(props: Props) => props.fullWidth ? '100%' : props.width};
  position: relative;
  font-family: ${FONTS.ROBOTO_REGULAR};
  font-size: 14px;

  ${(props: Props) => props.disabled && `
    pointer-events: none;

    ${TimePickerStyled} {
      opacity: 0.3;
    }
  `}

  ${TimePickerList} {
    margin-top: 40px;
  }
`;
