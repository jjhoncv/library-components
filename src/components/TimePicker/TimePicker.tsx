import * as React from 'react';
import * as PropTypes from 'prop-types';

import { useState, useEffect, useRef } from 'react';

import { Icon } from './../Icon';

import {
  TimePickerWrapper,
  TimePickerStyled,
  TimePickerInput,
  TimePickerIcon,
  TimePickerList,
  TimePickerItem
} from './styled';

interface Props {
  time?: string;
  width?: string;
  fullWidth?: boolean;
  error?: boolean;
  disabled?: boolean;
  getTimeSelected?: Function;
}

export const TimePicker: React.FC<Props> = ({
  time = '',
  width,
  fullWidth,
  error,
  disabled,
  getTimeSelected
}) => {

  const [ show, setShow ] = useState(false);
  const [ value, setValue ] = useState(time);
  const ref = useRef(null);

  useEffect(() => {
    document.addEventListener('mousedown', handleOutsideClick);
    return () => {
      document.removeEventListener('mousedown', handleOutsideClick);
    }
  })

  const handleOutsideClick = (e) => {
    const isOutside = !ref.current.contains(e.target);
    isOutside && setShow(false);
  }

  const TimePickerLisItem = ({setTime}) => {
    let items = [];
    for(let i=0; i<24; i++) {
      items.push(`${i}:00`);
      items.push(`${i}:30`);
    }

    const handleClick = (e) => {
      let time = e.target.innerText;
      setTime(time);
    }

    const formatTime = (time) => {
      return time.length == 4 ? `0${time}`: time;
    }

    return(
      <>
        {items.map((data, index) => (
          <TimePickerItem
            key={index}
            onClick={handleClick}
          >
            {formatTime(data)}
          </TimePickerItem>
        ))}
      </>
    );
  }

  const getTime = (time) => {
    setValue(time);
    setShow(false);

    getTimeSelected && getTimeSelected(time);
  }

  return (
    <TimePickerWrapper
      ref={ref}
      width={width}
      fullWidth={fullWidth}
      disabled={disabled}
    >
      <TimePickerStyled onClick={() => setShow(!show)}>
        <TimePickerInput
          type="text"
          placeholder="00:00"
          readOnly
          value={value}
          error={error}
        />
        <TimePickerIcon>
          <Icon name="chevron-down" color="#B6B6B6"/>
        </TimePickerIcon>
      </TimePickerStyled>
      { show && (
        <TimePickerList>
          <TimePickerLisItem setTime={getTime}/>
        </TimePickerList>
      )}
    </TimePickerWrapper>
  );
}

TimePicker.propTypes = {
  time: PropTypes.string,
  width: PropTypes.string,
  fullWidth: PropTypes.bool,
  error: PropTypes.bool,
  disabled: PropTypes.bool,
  getTimeSelected: PropTypes.func
}

TimePicker.defaultProps = {
  width: '200px'
}
