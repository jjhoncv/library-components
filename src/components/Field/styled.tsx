import styled from 'styled-components';
import { SelectWrapper } from '../Select/styled';
import { DatePickerWrapper } from '../DatePicker/styled';
import { TimePickerWrapper } from '../TimePicker/styled';
import { TextAreaStyled } from '../TextArea/styled';

import { COLORS, FONTS } from '../../ui';

interface GroupProps {
  keepProps?: boolean;
}

interface Props {
  errorMessage?: string;
  portionWidth?: number;
}

export const FieldStyled = styled.div`
  font-family: ${FONTS.ROBOTO_REGULAR};
  width: ${props => `${props.portionWidth * 100}%`};
  flex: 1 1 100%;


  input {
    width: 100%;
  }
  ${SelectWrapper} {
    width: 100%;
  }
  ${DatePickerWrapper} {
    width: 100%;
  }
  ${TimePickerWrapper} {
    width: 100%;
  }
  ${TextAreaStyled} {
    width: 100%;
  }
  margin-bottom: 14px;

  @media(min-width: 480px) {
    ${(props: Props) => props.portionWidth && `
      flex: 1 1 calc(${props.portionWidth * 100}% - 20px);
    `}
  }
`;

export const FieldGroup = styled.div`
  display: flex;
  flex-wrap: wrap;

  ${(props: GroupProps) => props.keepProps && `
    ${FieldStyled} {
      margin-right: 20px;
      flex: 1 1 calc(50% - 20px);
    }
    ${FieldStyled}:last-child {
      margin-right: 0;
    }
  `}

  @media(min-width: 480px) {
    ${FieldStyled} {
      margin-right: 20px;
    }
    ${FieldStyled}:last-child {
      margin-right: 0;
    }
  }
`;

export const Label = styled.label`
  display: block;
  margin-bottom: 2px;
  min-height: 19px;
  color: ${COLORS.BLACK};
  font-size: 12px;

  @media(min-width: 480px) {
    font-size: 14px;
  }
`;

export const BoxMessage = styled.span`
  display: block;
  font-size: 12px;
  line-height: 15px;
  margin-top: 4px;
  ${(props: Props) => props.errorMessage && `
    color: ${COLORS.ERROR_RED};
  `}
  height: 15px;
`;
