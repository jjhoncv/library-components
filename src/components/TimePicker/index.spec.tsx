import * as React from 'react';
import { shallow } from 'enzyme';
import { TimePicker } from './TimePicker';
import { TimePickerInput } from './styled';

describe('TimePicker Component', () => {
  describe('TimePicker => render snapshot', () => {
    it('Render correctly', () => {
      const wrapper = shallow(
        <TimePicker />
      );
      expect(wrapper.html()).toMatchSnapshot();
    })
  });

  describe('TimePicker Builder', () => {
    it('TimePicker => Render prop disabled', () => {
      const wrapper = shallow(
        <TimePicker disabled/>
      );
      expect(wrapper.prop('disabled')).toBe(true);
    });
    it('TimePicker => Render prop error', () => {
      const wrapper = shallow(
        <TimePicker error/>
      );
      expect(wrapper.find(TimePickerInput).prop('error')).toBe(true);
    });
    it('TimePicker => Render value input', () => {
      const wrapper = shallow(
        <TimePicker time="12:30"/>
      );
      expect(wrapper.find(TimePickerInput).prop('value')).toBe("12:30");
    });
  });
});
