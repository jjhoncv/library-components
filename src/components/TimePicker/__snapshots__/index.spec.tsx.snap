// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePicker Component TimePicker => render snapshot Render correctly 1`] = `"<div width=\\"200px\\" class=\\"sc-bZQynM hPPuJU\\"><div class=\\"sc-bwzfXH iyezzS\\"><input type=\\"text\\" placeholder=\\"00:00\\" readonly=\\"\\" value=\\"\\" class=\\"sc-htpNat jFAhRV\\"/><div class=\\"sc-bxivhb kzVfIo\\"><div color=\\"#B6B6B6\\" class=\\"sc-bdVaJa ilgoGI\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"><polyline points=\\"6 9 12 15 18 9\\"></polyline></svg></div></div></div></div>"`;
